doc/CMakeLists.txt

```diff
@@ -140,54 +138,11 @@ add_custom_target(
   COMMENT "Copying files to ${RST_OUT}"
 )
 
-# For incremental builds not to miss any source change this MUST be kept
-# a superset of INPUT= and FILE_PATTERNS= in zephyr.doxyfile.in
-#
-# NOTE: any changes here should be reflected in .github/workflows/doc-build.yml
-file(GLOB_RECURSE  DOXY_SOURCES
-  ${ZEPHYR_BASE}/include/*.[c,h,S]
-  ${ZEPHYR_BASE}/kernel/include/kernel_arch_interface.h
-  ${ZEPHYR_BASE}/lib/libc/*.[c,h,S]
-  ${ZEPHYR_BASE}/subsys/testsuite/ztest/include/*.[h,c,S]
-  ${ZEPHYR_BASE}/tests/*.[h,c,S]
-  )
-# For debug. Also find generated list in doc/_build/(build.ninja|CMakeFiles/)
-# message("DOXY_SOURCES= " ${DOXY_SOURCES})
-
-set(ARGS ${DOXYFILE_OUT})
-set(DOXY_RUN_TSTAMP ${CMAKE_CURRENT_BINARY_DIR}/last_doxy_run_tstamp)
-
-
-# Create timestamp first so we re-run if source files are edited while
-# doxygen is running
-add_custom_command(
-  OUTPUT ${DOXY_RUN_TSTAMP}
-  COMMAND cmake -E touch ${DOXY_RUN_TSTAMP}
-  COMMAND ${CMAKE_COMMAND}
-    -DCOMMAND=${DOXYGEN_EXECUTABLE}
-    -DARGS="${ARGS}"
-    -DOUTPUT_FILE=${DOXY_LOG}
-    -DERROR_FILE=${DOXY_LOG}
-    -DWORKING_DIRECTORY=${CMAKE_CURRENT_LIST_DIR}
-    -P ${ZEPHYR_BASE}/cmake/util/execute_process.cmake
-  DEPENDS ${DOXY_SOURCES}
-  COMMENT "Running ${DOXYGEN_EXECUTABLE}"
-)
-
-# Doxygen doesn't support incremental builds.
-# It could be ok because it's pretty fast.
-# But it's not because it has a cascade effect on sphinx:
-# https://sourceforge.net/p/doxygen/mailman/message/36580807/
-# For now this optimization speeds-up non-doxygen documentation work
-# only (by one order of magnitude).
 add_custom_target(
-  doxy_real_modified_times
-  COMMAND ${CMAKE_COMMAND} -E env
-    ${PYTHON_EXECUTABLE} _scripts/restore_modification_times.py
-    --loglevel  WARN  ${DOXY_OUT}/xml
+  doxygen
```

Comment:
Why are you keeping this target at all?

Reply:
Because Zephyr still publishes the standalone Doxygen documentation https://docs.zephyrproject.org/apidoc/latest/. I'm not sure though if `doxygen` target is used for that.

Reply:
OK, in that case, and if we want to be able to keep doing that, I think it's a plus point towards having CMake as a front end to Sphinx, because otherwise we are sort of duplicating the target here.

Reply:
Just checked, and no, it just assumes Doxygen `html` output is located at `_build/doxygen/html`. I've updated `doxyrunner` so that it also syncs `html`, this way CI is left unaffected.

Reply:
I'd maintain for now the `doxygen` target, there may be users that build the standalone Doxygen API pages.

Reply:
If we want to simplify the Doxygen standalone usage further, we could also create a regular Doxyfile by:

- Replacing `@ZEPHYR_BASE@` for `../`
- Replacing `@OUT_DIR@` for some pre-defined build directory, e.g. `_build/doxygen`

And always run Doxygen from the `doc` folder.

Reply:
this last suggestion seems quite reasonable, in order to be able to run doxygen locally.

Reply:
> If we want to simplify the Doxygen standalone usage further, we could also create a regular Doxyfile by:
> 
>     * Replacing `@ZEPHYR_BASE@` for `../`
> 

Actually, the CMake file still honors the ZEPHYR_BASE from the environment.
So if you start forcing Doxygen to always using `../` then doxygen might use another Zephyr base than rest of doc build that is invoked through CMake.

So if changing this, then the CMake file should be updated accordingly.

>     * Replacing `@OUT_DIR@` for some pre-defined build directory, e.g. `_build/doxygen`
> 
> 
> And always run Doxygen from the `doc` folder by just typing `doxygen`. I'd personally do that, CMake adds no value here.

When using CMake, users can specify their own build folder.
That means, I can do:
```
$ cd <path>/zephyr
$ cmake -GNinja -Bbuild_doc doc/
$ ninja -Cbuild_doc
```
so changing that will suddenly give me an extra `_build/doxygen` folder I need to consider.